clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -397,7 +397,7 @@ public void testStressfulSituation() throws Exception {
             List<ProducerBatch> batches = accum.drain(metadataCache, nodes, 5 * 1024, 0).get(node1.id());
             if (batches != null) {
                 for (ProducerBatch batch : batches) {
-                    for (Record record : batch.records().records())
+                    for (@SuppressWarnings("UnusedLocalVariable") Record ignored : batch.records().records())
                         read++;
```

Comment:
Why not use `ProducerBatch#recordCount` and avoid the loop entirely?

Reply:
I think the point of the test is actually simulating consumption of the batch in a scenario where multiple threads append to the accumulator. So we want to actually access the `MemoryRecords`, not just test the record counter.